x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -158,37 +161,24 @@ const getAgentAndPoliciesForEndpointsList = async (
 
   const policiesFound = (
     await sendBulkGetPackagePolicies(http, policyIdsToCheck)
-  ).items.reduce<PolicyIds>(
-    (list, packagePolicy) => {
-      list.packagePolicy[packagePolicy.id as string] = true;
-      list.agentPolicy[packagePolicy.id as string] = packagePolicy.policy_ids[0]; // TODO
-
-      return list;
-    },
-    { packagePolicy: {}, agentPolicy: {} }
-  );
+  ).items.reduce<NonExistingPolicies>((set, packagePolicy) => set.add(packagePolicy.id), new Set());
 
-  // packagePolicy contains non-existing packagePolicy ids whereas agentPolicy contains existing agentPolicy ids
-  const nonExistingPackagePoliciesAndExistingAgentPolicies = policyIdsToCheck.reduce<PolicyIds>(
-    (list, policyId: string) => {
-      if (policiesFound.packagePolicy[policyId as string]) {
-        list.agentPolicy[policyId as string] = policiesFound.agentPolicy[policyId];
-        return list;
+  const nonExistingPackagePolicies = policyIdsToCheck.reduce<NonExistingPolicies>(
```

Comment:
I am a bit confused about the change from Array to Set. How is it possible that you're doing `.reduce()` on Set? 

I bet I am missing something here, but to me this is just much more confusing

Reply:
it's possible, because `policyIdsToCheck` is an array : )
when calling `Array.prototype.reduce<T>()`, the generic type indicates the output type